pkg/pillar/cmd/volumemgr/handledownloader.go

```diff
@@ -121,36 +120,14 @@ func handleDownloaderStatusModify(ctxArg interface{}, key string,
 
 	// Handling even if Pending is set to process Progress updates
 
-	// XXX still need this downloader handshake?
-	// We handle two special cases in the handshake here
-	// 1. downloader added a status with RefCount=0 based on
-	// an existing file. We echo that with a config with RefCount=0
-	// 2. downloader set Expired in status when garbage collecting.
+	// We handle one special case in the handshake here, which is when
+	// downloader sets Expired in status when garbage collecting.
 	// If we have no RefCount we delete the config.
+	// If we have a config with non-zero RefCount it means there was
+	// a race and downloader will see the RefCount increase and clear the
+	// Expired flag (and not delete the file).
 
 	config := lookupDownloaderConfig(ctx, status.ObjType, status.ImageSha256)
```

Comment:
What is the reason to delete all of this? I never really used it - created `DownloaderConfig` from `downloadBlob()`- but I assumed there was a reason for it?

Reply:
This is old code from when downloader was running its own GC on /persist/downloads/*/pending. And the code was identical to the handshake with verifier for the GC. Downloader GC was removed a long time ago but not this code.